src/test_loader.rs

```diff
@@ -335,25 +336,33 @@ fn get_line_addresses<'data>(
         }
     }
 
+    let trace_analysis = add_line_analysis(analysis, config, &result);
+    result.merge(&trace_analysis);
+    Ok(result)
+}
+
+fn add_line_analysis(analysis: &HashMap<PathBuf, LineAnalysis>, config: &Config, input_trace: &TraceMap) -> TraceMap {
```

Comment:
It might not be a particularly hot path in the code, but we don't want to slow it down anyway when we can tidy it up a bit and keep the same performance (avoid the creation of two tracemaps both causing heap allocs and then a merge of them.

If you change the signature like so and don't create a tracemap inside this function (so add traces to `input_trace` instead of `result_trace`):

```suggestion
fn add_line_analysis(analysis: &HashMap<PathBuf, LineAnalysis>, config: &Config, input_trace: &mut TraceMap) {
```